Fix Box button not working after Star Pass

official-sounding opened the pull request:
When the skater changes positiosn from "Pivot" to "Jammer", the state is not correctly sync-ed between the PositionModel and the SkaterModel.  At Jam end, the Jammer position's PenaltyBox status is cleared, but then not re-set based on the SkaterModel.

This is now "fixed", but this PositionModel/SkaterModel nonsense is a mess, and in an ideal world I would just maintain the state on the SkaterModel and fire Position updates if the skater's status changes.  I would be more comfortable making those changes once unit tests are in place though (#72).

Closes #5

official-sounding committed:
Fix Box button not working after Star Pass

When the skater changes positiosn from "Pivot" to "Jammer", the state is not correctly sync-ed between the PositionModel and the SkaterModel.  At Jam end, the Jammer position's PenaltyBox status is cleared, but then not re-set based on the SkaterModel.

This is now "fixed", but this PositionModel/SkaterModel nonsense is a mess, and in an ideal world I would just maintain the state on the SkaterModel and fire Position updates if the skater's status changes.  I would be more comfortable making those changes once unit tests are in place though (#72).

Closes #5

brian-brazil commented:
Thanks for figuring this out, though I've not been fully through this bit of the code. The duplication of PositionModel/SkaterModel does sound messy.

I'm wondering if it's worth making this our first unittest.

official-sounding committed:
Merge branch 'dev' into bugfix/sp-box-button

official-sounding committed:
add "unit" tests for DefaultPositionModel

they aren't very unit-y, they depend rather heavily on the Team and Skater Models to work correctly, but I think this accomplishes good coverage of the Position interface.  Added Mockito to avoid pulling in too much of the universe to do these tests.  Near term plan (sometime in the few weeks) is to cook up a nice fixture to test the event system, add Tests for Skater and Team models, then start teasing apart the interweaving of the Team, Skater and Position models.

official-sounding commented:
I went ahead and added some "unit" tests covering base PositionModel functionality, and this scenario in particular.   I also introduced Mockito as a test dependency.

Despite the mocking, I end up having to make heavy use of TeamModel and SkaterModel in them to really test the position model functionality correctly, but it will at least start getting coverage to eventually allow for some separation.

brian-brazil commented on src/com/carolinarollergirls/scoreboard/defaults/DefaultPenaltyModel.java:
```diff
@@ -0,0 +1,27 @@
+import com.carolinarollergirls.scoreboard.Skater.Penalty;
+import com.carolinarollergirls.scoreboard.model.SkaterModel.PenaltyModel;
+
+public class DefaultPenaltyModel extends DefaultScoreBoardEventProvider implements PenaltyModel
```
Why did you move this out of the SkaterModel? Given PenaltyModel is a subinterface of SkaterModel I'd expect that to stay where it is.

brian-brazil commented on tests/com/carolinarollergirls/scoreboard/DefaultSkaterModelTests.java:
```diff
@@ -0,0 +1,14 @@
+
+	@Test
+	public void test() {
+		fail("Not yet implemented");
```
Will this cause the tests to fail?

brian-brazil reviewed (commented):
This looks generally good.

One thing I want to do at some point is hook things up so we can test the event state is right too.

official-sounding committed:
fix some issues with ant build with new tests

official-sounding commented:
Sorry, didn't mean to commit the skater model test file, obviously wasn't completed yet.  ditto splitting skatermodel/penaltymodel, and I'm cool keeping it where it is anyways.

after fixing those, I realized there were a few issues with build.xml, so I fixed those as well.

brian-brazil commented:
Great, thanks!
